lib/presentation/features/products/products_bloc.dart

```diff
@@ -48,20 +49,26 @@ class ProductsBloc extends Bloc<ProductsListEvent, ProductsState> {
       }
     } else if (event is ProductChangeSortRulesEvent) {
       yield state.getLoading();
-      final List<Product> filteredData = await productRepository.getProducts(
+      ProductsByFilterResult productResults = await findProductsByFilterUseCase.execute(
+        ProductsByFilterParams(
           categoryId: categoryId,
           filterRules: state.filterRules,
-          sortRules: event.sortBy);
+          sortBy: event.sortBy
+        ));
+      final List<Product> filteredData = productResults.products;
       yield state.copyWith(
         sortBy: event.sortBy,
         data: state.data.copyWith(filteredData),
       );
     } else if (event is ProductChangeFilterRulesEvent) {
       yield state.getLoading();
-      final List<Product> filteredData = await productRepository.getProducts(
+      ProductsByFilterResult productResults = await findProductsByFilterUseCase.execute(
+        ProductsByFilterParams(
           categoryId: categoryId,
-          filterRules: event.filterRules,
-          sortRules: state.sortBy);
+          filterRules: state.filterRules,
```

Comment:
why changing from event.filterRules to state.filterRules? The event was about changing the filter, but you just dismiss this data

Reply:
Changed back. Review latest commits.